src/target/riscv/riscv-013.c

```diff
@@ -280,13 +280,17 @@ static dm013_info_t *get_dm(struct target *target)
 	return dm;
 }
 
-static riscv_debug_reg_ctx_t get_riscv_debug_reg_ctx(struct target *target)
+static riscv_debug_reg_ctx_t get_riscv_debug_reg_ctx(const struct target *target)
 {
-	RISCV_INFO(r);
+	if (!target_was_examined(target)) {
+		const riscv_debug_reg_ctx_t default_context = {0};
```

Comment:
I feel like there should be a warning or at least a debug print about trying to get riscv_debug_reg_ctx before examination.

Reply:
Thank you for the review!
I would like to disagree.
The value of zero here means that the corresponding `is_set` bit is not set (since the values are not examined yet).
If a register decoding requires a value which is not set from the context, it would fail with an assertion.
An example of such an assertion can be seen here: https://github.com/riscv/riscv-openocd/blob/riscv/src/target/riscv/debug_defines.c#L178

Reply:
Alright then. Thanks for the clearup.